modules/account-lib/src/coin/hbar/keyPair.ts

```diff
@@ -53,7 +55,10 @@ export class KeyPair extends Ed25519KeyPair {
   /** @inheritdoc */
   recordKeysFromPrivateKeyInProtocolFormat(prv: string): DefaultKeys {
     try {
-      const hederaPrv = Ed25519PrivateKey.fromString(prv);
+      if (prv.length % 2 !== 0) {

```

Comment:
```suggestion
      if (/^([a-f0-9]{2})+$/i.test(prv)) {
```

This might be a bit more what you're after if you want to make sure it's hex (remove the `i` at the end of the regex if you want this to be case sensitive)

Reply:
Also, pull this check above the try/catch so you don't catch this locally thrown exception.